model/task/task_test.go

```diff
@@ -960,6 +960,35 @@ func TestBlockedState(t *testing.T) {
 	assert.Equal("blocked", state)
 }
 
+func TestBlockedStateForExecTasks(t *testing.T) {
```

Comment:
It shouldn't make a difference but could you also add a test case where the task that's blocking things is outside of the display task?